AlphaWallet/Activities/ActivitiesService.swift

```diff
@@ -242,8 +242,10 @@ class ActivitiesService: NSObject, ActivitiesServiceType {
 
             strongSelf.reloadViewController(reloadImmediately: reloadImmediately)
 
-            for (activity, tokenObject, tokenHolder) in activitiesAndTokens {
-                strongSelf.refreshActivity(tokenObject: tokenObject, tokenHolder: tokenHolder, activity: activity)
+            DispatchQueue.main.async {
+                for (activity, tokenObject, tokenHolder) in activitiesAndTokens {
+                    strongSelf.refreshActivity(tokenObject: tokenObject, tokenHolder: tokenHolder, activity: activity)
```

Comment:
This might potentionally have many refreshes on the main thread though. Does it block anything?

Reply:
updated it, `refreshActivity` can be updated on `.background` queue, just locked access to non atomic `activities` where the app was crashing